build.py

```diff
@@ -238,7 +241,9 @@ def build_bess():
     print 'Building BESS daemon...'
     cmd('bin/bessctl daemon stop 2> /dev/null || true')
     cmd('rm -f core/bessd')     # force relink as DPDK might have been rebuilt
-    cmd('make -C core')
+    if cxx_flags or ld_flags:
+        cmd('make -C core CXXFLAGS=%s CFLAGS=%s LDFLAGS=%s' %\
```

Comment:
One concern with this is that once `build.py` is run with custom flags, subsequent builds without `build.py` (i.e., running `make` in the `core` directory) will fail. This could be problematic since during development phase developers should be able to use the `Makefile` without without dealing with `build.py`. Somehow the script (which is currently like a poor man's autoconf) should remember special flags that were used during the initial build.

The `core/extra.mk` file is supposed to be used for this purpose, although the current usage is limited for additional libraries required by DPDK build.